src/System.Management.Automation/engine/Modules/ModuleUtils.cs

```diff
@@ -79,22 +93,30 @@ internal static IEnumerable<string> GetAllAvailableModuleFiles(string topDirecto
                 catch (IOException) { }
                 catch (UnauthorizedAccessException) { }
 
-                var files = Directory.GetFiles(directoryToCheck, "*", SearchOption.TopDirectoryOnly);
-                foreach (string moduleFile in files)
+                fileAtIndexWasReturned.Clear();
```

Comment:
Is there a chance the `HashSet` won't be empty at this point? If it needs to be guaranteed empty, maybe it's worth moving the initialisation down to above the `foreach` where it's used

Reply:
Not sure I understand your comment. This is to make sure it's empty when we start to process all the files from the directory.

Reply:
Just mean that there don't seem to be any references to `fileAtIndexWasReturned` before the call to `Clear()` — so is there a need to call `Clear()`?

If there's a strong need to ensure that `fileAtIndexWasReturned` is empty when it's used later, then maybe it should just be defined and created later, above the foreach loop.